server/test/services/codeactions/specifyCompilerVersion.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from "fs";
+import * as path from "path";
+import { SpecifyCompilerVersion } from "@compilerDiagnostics/diagnostics/SpecifyCompilerVersion";
+import { assertCodeAction } from "./asserts/assertCodeAction";
+
+describe("Code Actions", () => {
+  const specifyCompilerVersion = new SpecifyCompilerVersion();
+  let testContractText: string;
+
+  before(async () => {
+    testContractText = (
+      await fs.promises.readFile(
+        path.join(__dirname, "testData", "SpecifyCompilerVersion.sol")
+      )
+    ).toString();
+  });
+
+  describe("Specify compiler version", () => {
+    it("adds pragma solidity line", async () => {
```

Comment:
Nice, could we get two further tests, so we end up with:

* no spdx license line
* a spdx license line
* no spdx license line, but a standard comment on the first line

That way if someone is trying to figure out whether the behaviour around add pragma lines when the first line is a comment vs a license line, we have an explicit test to demonstrate what we intended the rule to be.